src/main/java/org/killbill/billing/plugin/dwolla/api/DwollaPaymentPluginApi.java

```diff
@@ -173,13 +247,12 @@ private PaymentTransactionInfoPlugin createDwollaTransaction(final TransactionTy
             );
 
         } catch (ApiException e) {
-            // TODO parse error code
-            final String errorCode = "";
-            final String errorMsg = e.getMessage();
-
+            final DwollaErrorResponse error = JsonHelper.getObjectFromRequest(e.getMessage(), DwollaErrorResponse.class);
+            final String errorCode = error.get_embedded().get("errors").get(0).get("code").toString();
```

Comment:
Maybe check the `errors` list has at least one element to avoid throwing another exception here.